src/main/java/com/google/cloud/spanner/hibernate/SpannerTableExporter.java

```diff
@@ -50,6 +50,12 @@ public SpannerTableExporter(SpannerDialect spannerDialect) {
     /* The current implementation does not support interleaved tables for collections
      * or UNIQUE constraints/indexes for relationships
      * */
+
+    if (!table.hasPrimaryKey()) {
+      throw new UnsupportedOperationException("Cloud Spanner requires tables and entities to have"
```

Comment:
I think this is more of an `IllegalArgumentException`. UOE is usually for completely unimplemented methods.

Reply:
I think that case is better covered by the `UnimplementedException`. Here I think it really is an unsupported operation because Hibernate assumes not having a PK is a valid thing to create because standard SQL DBs accept it and it is generally a valid operation to create such a table (vs. generally illegal), but we cannot support it.